File: app/com/github/fdesu/data/repo/CrudRepo.java
package com.github.fdesu.data.repo;

import java.util.stream.Stream;


public interface CrudRepo<T, I> {

    T findById(I id);

    Stream<T> all();

    I persist(T entity);

    T update(T entity);

    void delete(I id);

}

File: app/com/github/fdesu/data/repo/CrudRepo.scala
package com.github.fdesu.data.repo

import java.util.stream

trait CrudRepo[T, I] {

  def findById(id: I): T

  def all: stream.Stream[T]

  def persist(entity: T): I

  def update(entity: T): T

  def delete(id: I): Unit

}
